api/v1alpha1/performanceprofile_conversion.go

```diff
@@ -0,0 +1,120 @@
+package v1alpha1
+
+import (
+	"github.com/openshift-kni/performance-addon-operators/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this PerformanceProfile to the Hub version (v1).
+func (curr *PerformanceProfile) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1.PerformanceProfile)
+
+	// ObjectMeta
+	dst.ObjectMeta = curr.ObjectMeta
+
+	// Spec
+	dst.Spec.CPU = new(v1.CPU)
+	hugePagesSize := v1.HugePageSize(*curr.Spec.HugePages.DefaultHugePagesSize)
+	reserved := v1.CPUSet(*curr.Spec.CPU.Reserved)
+	dst.Spec.CPU.Reserved = &reserved
+	isolated := v1.CPUSet(*curr.Spec.CPU.Isolated)
+	dst.Spec.CPU.Isolated = &isolated
+	dst.Spec.CPU.BalanceIsolated = curr.Spec.CPU.BalanceIsolated
+
+	dst.Spec.HugePages = new(v1.HugePages)
+	dst.Spec.HugePages.DefaultHugePagesSize = &hugePagesSize
+	dst.Spec.HugePages.Pages = make([]v1.HugePage, len(curr.Spec.HugePages.Pages))
+	for i, p := range curr.Spec.HugePages.Pages {
+		dst.Spec.HugePages.Pages[i] = v1.HugePage{
+			Size: v1.HugePageSize(p.Size), Count: p.Count, Node: p.Node,
+		}
+	}
+
+	dst.Spec.MachineConfigLabel = make(map[string]string)
+	for k, v := range curr.Spec.MachineConfigLabel {
+		dst.Spec.MachineConfigLabel[k] = v
+	}
+
+	dst.Spec.MachineConfigPoolSelector = make(map[string]string)
+	for k, v := range curr.Spec.MachineConfigPoolSelector {
+		dst.Spec.MachineConfigPoolSelector[k] = v
+	}
+
+	dst.Spec.NodeSelector = make(map[string]string)
+	for k, v := range curr.Spec.NodeSelector {
+		dst.Spec.NodeSelector[k] = v
+	}
+
+	dst.Spec.RealTimeKernel = new(v1.RealTimeKernel)
+	dst.Spec.RealTimeKernel.Enabled = curr.Spec.RealTimeKernel.Enabled
+
+	dst.Spec.AdditionalKernelArgs = curr.Spec.AdditionalKernelArgs
+
+	dst.Spec.NUMA = new(v1.NUMA)
+	dst.Spec.NUMA.TopologyPolicy = curr.Spec.NUMA.TopologyPolicy
+
+	// Status
+	dst.Status.Conditions = curr.Status.Conditions
+	dst.Status.Tuned = curr.Status.Tuned
+	dst.Status.RuntimeClass = curr.Status.RuntimeClass
+
+	// +kubebuilder:docs-gen:collapse=rote conversion
+	return nil
+}
+
+// ConvertFrom converts from the Hub version (v1) to this version.
+func (curr *PerformanceProfile) ConvertFrom(srcRaw conversion.Hub) error {
```

Comment:
all problems specified under the `ConvertTo` also relevant for `ConvertFrom`

Reply:
done.
BTW, I was under impression a shallow copy here would be enough the src object is not suppose to diverge.
We have our own conversion server that gets a profile from the api and returns another one to the caller.
So even if we get a pointer to the profile, semantically is ok to use it.

But better safe than sorry so I made the changes.